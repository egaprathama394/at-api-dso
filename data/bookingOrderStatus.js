export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
