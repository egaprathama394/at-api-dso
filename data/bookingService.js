const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
