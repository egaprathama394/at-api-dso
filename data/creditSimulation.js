export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};