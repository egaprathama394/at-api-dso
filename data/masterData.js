
export default {
  companyCode: '0003',
  masterDataInfo: [
    {
      lastSync: '1900-01-01 00:00:00',
      objectName: 'Branch',
    },
  ],
  vehicleBrandCode: 'DAIHATSU',
  accountType: '1',
  action: '',
  deviceType: '0',
  operatingSystem: '0',
  serVersion: '1.0.0',
};

export const JobServiceCompany = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'JobService_Company',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const VehicleBrand = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'vehicleBrand',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const district = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'district',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const province = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'province',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const region = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'region',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const holidayCalendar = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'holidayCalendar',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

// looping data for job service company

export const loopVehicleType = [
  {
    vehicleType: 'AYLA',
    title: 'test case get master data job service company where vehicle = AYLA',
  },
  {
    vehicleType: 'GRANMAX',
    title: 'test case get master data job service company where vehicle = GRANMAX',
  },
  {
    vehicleType: 'GRANMAX1.3',
    title: 'test case get master data job service company where vehicle = GRANMAX1.3',
  },
  {
    vehicleType: 'GRANMAX1.5',
    title: 'test case get master data job service company where vehicle = GRANMAX1.5',
  },
  {
    vehicleType: 'LUXIO',
    title: 'test case get master data job service company where vehicle = LUXIO',
  },
  {
    vehicleType: 'Sigra 1.0',
    title: 'test case get master data job service company where vehicle = Sigra 1.0',
  },
  {
    vehicleType: 'Sigra 1.2',
    title: 'test case get master data job service company where vehicle = Sigra 1.2',
  },
  {
    vehicleType: 'SIRION',
    title: 'test case get master data job service company where vehicle = SIRION',
  },
  {
    vehicleType: 'TERIOS',
    title: 'test case get master data job service company where vehicle = TERIOS',
  },
  {
    vehicleType: 'XENIA1.0',
    title: 'test case get master data job service company where vehicle = XENIA1.0',
  },
  {
    vehicleType: 'XENIA1.3',
    title: 'test case get master data job service company where vehicle = XENIA1.3',
  },
  {
    vehicleType: 'OTHERS',
    title: 'test case get master data job service company where vehicle = OTHERS',
  },
];

export const VehicleTestDrive = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'VehicleTestDrive',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const parameterConfigApp = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'ParameterConfigApplication',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
