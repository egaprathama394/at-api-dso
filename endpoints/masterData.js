/* eslint no-unused-vars: off */
/* global env */
const supertest = require('supertest');
const env = require('dotenv').config();

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;

const api = supertest(process.env.API_BASE_URL_MOBILE);

export default body => api.post('/api/masterdatainfo')
  .set('ClientTag', process.env.CLIENT_TAG)
  .set('server-code', process.env.SERVER_CODE_MOBILE)
  .send(body);
