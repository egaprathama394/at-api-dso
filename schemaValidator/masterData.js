export default {
  properties: {
    acknowledge: {
      type: 'number',
    },
    branch: {
      type: 'array',
      properties: {
        id: {
          type: 'number',
        },
        companyCode: {
          type: 'string',
        },
        regionName: {
          type: 'string',
        },
        branchCode: {
          type: 'string',
        },
        branchTypeCode: {
          type: 'string',
        },
        branchDescription: {
          type: 'string',
        },
        branchPICName: {
          type: 'string',
        },
        bookType: {
          type: 'number',
        },
        postalCode: {
          type: 'string',
        },
        address: {
          type: 'string',
        },
        latitude: {
          type: 'number',
        },
        longitude: {
          type: 'number',
        },
        distance: {
          type: 'number',
        },
        phone1: {
          type: 'string',
        },
        phone2: {
          type: 'string',
        },
        fax: {
          type: 'string',
        },
        branchEmailTo: {
          type: 'string',
        },
        branchEmailCc: {
          type: 'string',
        },
        webSiteLink: {
          type: 'string',
        },
        mondayToFriday: {
          type: 'string',
        },
        saturday: {
          type: 'string',
        },
        sunday: {
          type: 'string',
        },
        isMainDealer: {
          type: 'boolean',
        },
        isActive: {
          type: 'boolean',
        },
        districtId: {
          type: 'number',
        },
        districtName: {
          type: 'string',
        },
        hasBookingStall: {
          type: 'boolean',
        },
      },
      required: [
        'id',
        'companyCode',
        'regionName',
        'branchCode',
        'branchTypeCode',
        'branchDescription',
        'branchPICName',
        'bookType',
        'postalCode',
        'address',
        'latitude',
        'longitude',
        'distance',
        'phone1',
        'phone2',
        'fax',
        'branchEmailTo',
        'branchEmailCc',
        'webSiteLink',
        'mondayToFriday',
        'saturday',
        'sunday',
        'isMainDealer',
        'isActive',
        'districtId',
        'districtName',
        'hasBookingStall',
      ],
    },
    build: {
      type: 'string',
    },
    district: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    version: {
      type: 'string',
    },
    location: {
      type: 'array',
    },
  },
  required: [
    'acknowledge',
    'branch',
    'build',
    'district',
    'hobbies',
    'holidayCalendar',
    'jobService',
    'jobServiceCompany',
    'parameterConfigApplication',
    'province',
    'region',
    'vehicleBrand',
    'vehicleTestDrive',
    'version',
    'location',
  ],
  additionalProperties: false,
};

export const jobServiceCompanySchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'jobCode',
          'jobDescription',
          'actual',
          'rate',
          'price',
          'vehicleType',
          'serviceCategory',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          jobCode: {
            type: 'string',
          },
          jobDescription: {
            type: 'string',
          },
          actual: {
            type: 'number',
          },
          rate: {
            type: 'integer',
          },
          price: {
            type: 'integer',
          },
          vehicleType: {
            type: 'string',
          },
          serviceCategory: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const vehicleBrandSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'vehicleBrandCode',
          'vehicleType',
          'vehicleDescription',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          vehicleBrandCode: {
            type: 'string',
          },
          vehicleType: {
            type: 'string',
          },
          vehicleDescription: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const masterDataProvinceSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },

  },
  holidayCalendar: {
    type: 'array',
  },
  parameterConfigApplication: {
    type: 'array',
  },
  hobbies: {
    type: 'array',
  },
  province: {
    type: 'array',
    items: {
      type: 'object',
      required: [
        'id',
        'provinceCode',
        'provinceName',
        'isActive',
      ],
      properties: {
        id: {
          type: 'integer',
        },
        provinceCode: {
          type: 'string',
        },
        provinceName: {
          type: 'string',
        },
        isActive: {
          type: 'boolean',
        },
      },

    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const districtSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'provinceId',
          'districtCode',
          'districtName',
          'isActive',
          'regions',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          provinceId: {
            type: 'integer',
          },
          districtCode: {
            type: 'string',
          },
          districtName: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
          regions: {
            type: 'array',
          },
        },
      },
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const RegionsSchema = {
  type: 'array',
  items: {
    type: 'object',
    required: [
      'id',
      'salesOrganization',
      'regionCode',
      'regionName',
    ],
    properties: {
      id: {
        type: 'integer',
      },
      salesOrganization: {
        type: 'string',
      },
      regionCode: {
        type: 'string',
      },
      regionName: {
        type: 'string',
      },
    },
  },
};


export const masterDataRegionSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'salesOrganization',
          'regionCode',
          'regionName',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          salesOrganization: {
            type: 'string',
          },
          regionCode: {
            type: 'string',
          },
          regionName: {
            type: 'string',
          },
        },
      },
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const holidayCalendarSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'holidayDate',
          'holidayDescription',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          holidayDate: {
            type: 'string',
          },
          holidayDescription: {
            type: 'string',
          },
        },
      },
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const vehicleTestDriveSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'vehicleBrandId',
          'vehicleType',
          'branchId',
          'branchCode',
          'isTestDrive',
          'isActive',
          'modelMaterialId',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          vehicleBrandId: {
            type: 'integer',
          },
          vehicleType: {
            type: 'string',
          },
          branchId: {
            type: 'integer',
          },
          branchCode: {
            type: 'string',
          },
          isTestDrive: {
            type: 'boolean',
          },
          isActive: {
            type: 'boolean',
          },
          modelMaterialId: {
            type: 'integer',
          },
        },
      },
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const parameterConfigApplicationSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'companyCode',
          'key',
          'value',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          companyCode: {
            type: 'string',
          },
          key: {
            type: 'string',
          },
          value: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};
