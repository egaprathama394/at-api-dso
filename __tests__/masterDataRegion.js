/* eslint-disable max-len */
/* eslint-disable eqeqeq */
/* eslint-disable no-shadow */
import getMasterData from '../endpoints/masterData';
import * as masterDataSchema from '../schemaValidator/masterData';
import * as body from '../data/masterData';

describe('Master Data Region', () => {
  test('Can get master data Region successfully - positive test cases', async () => {
    const response = await getMasterData(body.region);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(masterDataSchema.masterDataRegionSchema);
    expect(response.body.region.length).toEqual(44);
    expect(response.body).toMatchSnapshot();
  });
});
