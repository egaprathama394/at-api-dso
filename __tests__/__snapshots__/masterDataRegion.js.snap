// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master Data Region Can get master data Region successfully - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "branch": Array [],
  "build": "0",
  "district": Array [],
  "hobbies": Array [],
  "holidayCalendar": Array [],
  "jobService": Array [],
  "jobServiceCompany": Array [],
  "location": Array [],
  "parameterConfigApplication": Array [],
  "province": Array [],
  "region": Array [
    Object {
      "id": 42,
      "regionCode": "D01",
      "regionName": "DKI Jakarta",
      "salesOrganization": "0003",
    },
    Object {
      "id": 44,
      "regionCode": "D02",
      "regionName": "Jawa Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 45,
      "regionCode": "D03",
      "regionName": "Jawa Tengah",
      "salesOrganization": "0003",
    },
    Object {
      "id": 46,
      "regionCode": "D04",
      "regionName": "Jawa Timur",
      "salesOrganization": "0003",
    },
    Object {
      "id": 47,
      "regionCode": "D05",
      "regionName": "DI Yogyakarta",
      "salesOrganization": "0003",
    },
    Object {
      "id": 48,
      "regionCode": "D06",
      "regionName": "NAD",
      "salesOrganization": "0003",
    },
    Object {
      "id": 49,
      "regionCode": "D07",
      "regionName": "Sumatera Utara",
      "salesOrganization": "0003",
    },
    Object {
      "id": 50,
      "regionCode": "D08",
      "regionName": "Sumatera Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 51,
      "regionCode": "D09",
      "regionName": "Riau",
      "salesOrganization": "0003",
    },
    Object {
      "id": 52,
      "regionCode": "D10",
      "regionName": "Jambi",
      "salesOrganization": "0003",
    },
    Object {
      "id": 53,
      "regionCode": "D11",
      "regionName": "Sumatera Selatan",
      "salesOrganization": "0003",
    },
    Object {
      "id": 54,
      "regionCode": "D12",
      "regionName": "Bengkulu",
      "salesOrganization": "0003",
    },
    Object {
      "id": 55,
      "regionCode": "D13",
      "regionName": "Lampung",
      "salesOrganization": "0003",
    },
    Object {
      "id": 56,
      "regionCode": "D14",
      "regionName": "Kalimantan Selatan",
      "salesOrganization": "0003",
    },
    Object {
      "id": 57,
      "regionCode": "D15",
      "regionName": "Kalimantan Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 58,
      "regionCode": "D16",
      "regionName": "Kalimantan Tengah",
      "salesOrganization": "0003",
    },
    Object {
      "id": 59,
      "regionCode": "D17",
      "regionName": "Kalimantan Timur",
      "salesOrganization": "0003",
    },
    Object {
      "id": 60,
      "regionCode": "D18",
      "regionName": "Sulawesi Selatan",
      "salesOrganization": "0003",
    },
    Object {
      "id": 61,
      "regionCode": "D19",
      "regionName": "Sulawesi Tenggara",
      "salesOrganization": "0003",
    },
    Object {
      "id": 62,
      "regionCode": "D20",
      "regionName": "Sulawesi Tengah",
      "salesOrganization": "0003",
    },
    Object {
      "id": 63,
      "regionCode": "D21",
      "regionName": "Sulawesi Utara",
      "salesOrganization": "0003",
    },
    Object {
      "id": 64,
      "regionCode": "D22",
      "regionName": "Bali",
      "salesOrganization": "0003",
    },
    Object {
      "id": 65,
      "regionCode": "D23",
      "regionName": "Nusa Tenggara Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 66,
      "regionCode": "D24",
      "regionName": "Nusa Tenggara Timur",
      "salesOrganization": "0003",
    },
    Object {
      "id": 67,
      "regionCode": "D25",
      "regionName": "Maluku",
      "salesOrganization": "0003",
    },
    Object {
      "id": 68,
      "regionCode": "D26",
      "regionName": "Papua",
      "salesOrganization": "0003",
    },
    Object {
      "id": 69,
      "regionCode": "D27",
      "regionName": "Timor Timur",
      "salesOrganization": "0003",
    },
    Object {
      "id": 70,
      "regionCode": "D28",
      "regionName": "Banten",
      "salesOrganization": "0003",
    },
    Object {
      "id": 71,
      "regionCode": "D29",
      "regionName": "Irian Jaya Timur",
      "salesOrganization": "0003",
    },
    Object {
      "id": 72,
      "regionCode": "D30",
      "regionName": "Irian Jaya Tengah",
      "salesOrganization": "0003",
    },
    Object {
      "id": 73,
      "regionCode": "D31",
      "regionName": "Maluku Utara",
      "salesOrganization": "0003",
    },
    Object {
      "id": 74,
      "regionCode": "D32",
      "regionName": "Kep. Bangka Belitung",
      "salesOrganization": "0003",
    },
    Object {
      "id": 75,
      "regionCode": "D33",
      "regionName": "Kepulauan Riau",
      "salesOrganization": "0003",
    },
    Object {
      "id": 76,
      "regionCode": "D34",
      "regionName": "Sulawesi Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 77,
      "regionCode": "D35",
      "regionName": "Gorontalo",
      "salesOrganization": "0003",
    },
    Object {
      "id": 78,
      "regionCode": "D36",
      "regionName": "Irian Jaya Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 79,
      "regionCode": "D37",
      "regionName": "Papua Barat",
      "salesOrganization": "0003",
    },
    Object {
      "id": 80,
      "regionCode": "D364",
      "regionName": "DSO Yogya Mataram",
      "salesOrganization": "0003",
    },
    Object {
      "id": 81,
      "regionCode": "D066",
      "regionName": "DSO BSD",
      "salesOrganization": "0003",
    },
    Object {
      "id": 144,
      "regionCode": "D38",
      "regionName": "DKI 1",
      "salesOrganization": "0003",
    },
    Object {
      "id": 146,
      "regionCode": "D39",
      "regionName": "DKI 2",
      "salesOrganization": "0003",
    },
    Object {
      "id": 147,
      "regionCode": "40",
      "regionName": "Kalimantan",
      "salesOrganization": "0003",
    },
    Object {
      "id": 148,
      "regionCode": "41",
      "regionName": "Sumatera",
      "salesOrganization": "0003",
    },
    Object {
      "id": 149,
      "regionCode": "42",
      "regionName": "IBT",
      "salesOrganization": "0003",
    },
  ],
  "vehicleBrand": Array [],
  "vehicleTestDrive": Array [],
  "version": "1.0",
}
`;
